lib/Server.js

```diff
@@ -444,16 +444,49 @@ class Server {
 
     const compilerOptions = this.getCompilerOptions();
     // TODO remove `{}` after drop webpack v4 support
-    const watchOptions = compilerOptions.watchOptions || {};
-    const defaultWatchOptions = {
-      ignoreInitial: true,
-      persistent: true,
-      followSymlinks: false,
-      atomic: false,
-      alwaysStat: true,
-      ignorePermissionErrors: true,
-      ...watchOptions,
+    const compilerWatchOptions = compilerOptions.watchOptions || {};
+
+    const getDefaultWatchOptions = (watchOptions = {}) => {
+      // duplicate the same massaging of options that watchpack performs
+      // https://github.com/webpack/watchpack/blob/master/lib/DirectoryWatcher.js#L49
+      // this isn't an elegant solution, but we'll improve it in the future
+      const usePolling =
+        typeof watchOptions.usePolling !== "undefined"
+          ? watchOptions.usePolling
+          : Boolean(watchOptions.poll);
+
+      const interval =
+        // eslint-disable-next-line no-nested-ternary
+        typeof watchOptions.interval !== "undefined"
+          ? watchOptions.interval
+          : typeof watchOptions.poll === "number"
+          ? watchOptions.poll
+          : // eslint-disable-next-line no-undefined
+            undefined;
+
+      const finalWatchOptions = {
+        ignoreInitial: true,
+        persistent: true,
+        followSymlinks: false,
+        atomic: false,
+        alwaysStat: true,
+        ignorePermissionErrors: true,
+        ...compilerWatchOptions,
+        ...watchOptions,
+        ignored: watchOptions.ignored,
+        usePolling,
+        interval,
+      };
+
+      if (finalWatchOptions.poll) {
+        delete finalWatchOptions.poll;
+      }
```

Comment:
Avoid it, you already know about `poll` is not valid for chokidar, only pass valid options

Reply:
`poll` can also come from `compilerWatchOptions`.

Reply:
Just not put `poll` in your default options